asb-native/src/main/java/org/ballerinalang/asb/MessageDispatcher.java

```diff
@@ -0,0 +1,217 @@
+    private String connectionKey;
+    private BRuntime runtime;
+    private IMessageReceiver receiver;
+    private static final StrandMetadata ON_MESSAGE_METADATA = new StrandMetadata(ORG_NAME, ASB,
+            ASB_VERSION, FUNC_ON_MESSAGE);
+
+    /**
+     * Initialize the Message Dispatcher.
+     *
+     * @param service Ballerina service instance.
+     * @param runtime Ballerina runtime instance.
+     * @param iMessageReceiver Asb MessageReceiver instance.
+     */
+    public MessageDispatcher(BObject service, BRuntime runtime, IMessageReceiver iMessageReceiver) {
+        this.service = service;
+        this.queueName = getQueueNameFromConfig(service);
+        this.connectionKey = getConnectionStringFromConfig(service);
+        this.runtime = runtime;
+        this.receiver = iMessageReceiver;
+    }
+
+    /**
+     * Get queue name from annotation configuration attached to the service.
+     *
+     * @param service Ballerina service instance.
+     * @return Queue name from annotation configuration attached to the service.
+     */
+    public static String getQueueNameFromConfig(BObject service) {
+        BMap serviceConfig = (BMap) ((AnnotatableType) service.getType())
+                .getAnnotation(BStringUtils.fromString(AsbConstants.PACKAGE_RABBITMQ_FQN + ":"
+                        + AsbConstants.SERVICE_CONFIG));
+        @SuppressWarnings(AsbConstants.UNCHECKED)
+        BMap<BString, Object> queueConfig =
+                (BMap) serviceConfig.getMapValue(AsbConstants.ALIAS_QUEUE_CONFIG);
+        return queueConfig.getStringValue(AsbConstants.QUEUE_NAME).getValue();
+    }
+
+    /**
+     * Get connection string from annotation configuration attached to the service.
+     *
+     * @param service Ballerina service instance.
+     * @return Connection string from annotation configuration attached to the service.
+     */
+    public static String getConnectionStringFromConfig(BObject service) {
+        BMap serviceConfig = (BMap) ((AnnotatableType) service.getType())
+                .getAnnotation(BStringUtils.fromString(AsbConstants.PACKAGE_RABBITMQ_FQN + ":"
+                        + AsbConstants.SERVICE_CONFIG));
+        @SuppressWarnings(AsbConstants.UNCHECKED)
+        BMap<BString, Object> queueConfig =
+                (BMap) serviceConfig.getMapValue(AsbConstants.ALIAS_QUEUE_CONFIG);
+        return queueConfig.getStringValue(CONNECTION_STRING).getValue();
+    }
+
+    /**
+     * Start receiving messages asynchronously and dispatch the messages to the attached service.
+     *
+     * @param listener Ballerina listener object.
+     */
+    public void receiveMessages(BObject listener) {
+        LOG.info("Consumer service started for queue " + queueName);
+
+        try {
+            ExecutorService executorService = Executors.newSingleThreadExecutor();
+            this.pumpMessage(receiver, executorService);
+            LOG.info("\tDone receiving messages from \n" + receiver.getEntityPath());
```

Comment:
Having Info logs for every message is not required and may affect the performance.

Reply:
Working on removing Info logs

Reply:
removed unwanted Info logs